audbackend/core/artifactory.py

```diff
@@ -431,6 +438,7 @@ def _use_legacy_file_structure(
             self,
             *,
             extensions: typing.List[str] = None,
+            regex: bool = False,
```

Comment:
What happens if you mix regex based extensions with non-regex based, e.g. `extensions=['\d+.tar.gz', 'tar.gz']`?

Reply:
I'm asking as it seems a little bit critical, that we can provide a list with extensions, some using a regex pattern others not, but we can only provide a single `regex=True`.

Reply:
You are not supposed to mix. Either you provide all as regex or none.